app/forms/openid_connect_authorize_form.rb

```diff
@@ -68,6 +66,22 @@ def service_provider
     @_service_provider ||= ServiceProvider.from_issuer(client_id)
   end
 
+  def link_identity_to_client_id(current_user, rails_session_id)
```

Comment:
I think what we have now works. It's the Identity model that's being created/updated, and we call a class called `IdentityLinker`.